docs/docs/recognizers/strategy.md

```diff
@@ -0,0 +1,50 @@
+# Strategy
+## Introduction
+
+### Context of use
+The strategy pattern comes in handy when multiple methods do the same thing in different ways, this pattern is in this case used to reduce code duplication. The gang of four describes the Strategy Pattern in the following manner: “Define a family of algorithms, encapsulate each one, and make them interchangeable. Strategy lets the algorithm vary independently from clients that use it”. It enables the user to use different business rules or algorithms depending on the context in which they occur.
+
+### Example of use
+An example case that shows the need of the pattern is when someone uses a route calculator which calculates the shortest route from A to B. Someone then might want to use the strategy pattern for each of the different routing options (Bike, Car, etc). Then the code can just call a function CreateRoute independent of which routing algorithm is selected.
+
+## Explanation of Architecture
+![UML Strategy](https://refactoring.guru/images/patterns/diagrams/strategy/structure.png)
+
+The **Context** is the class from where the code gets executed. It maintains a reference to one of the concrete strategies and communicates with the object only via the Strategy interface.
+
+The **Strategy** is an interface. This interface needs to contain a method that will be used to execute the algorithm as defined in the ConcreteStrategie classes.
```

Comment:
Don't you mean: "This interface needs to contain a method that will be used to execute the algorithm implemented in the ConcreteStrategie classes."

Reply:
The interface Strategy could also be an abstract class so I will change it to inherits/implements. In addition to this I think I did not stated that this could be the case so I will add that too.